### ISSUE TITLE
IaaS ADR for transparency on upstream DNS

### ISSUE THREAD

**Issue by garloff** [SovereignCloudStack/issues#230]
As a privacy conscious user, I want to know which public DNS servers are used by my operator.

## Definition of Ready:

- [ ] User Story is small enough to be finished within one sprint
- [ ] User Story is clear and understood by the whole team
- [ ] Acceptance criteria are defined
- [ ] Acceptance criteria are clear and understood by the whole team

## Definition of Done:

- [ ] All acceptance criteria are met
- [ ] Changes have been reviewed
- [ ] CI tests have run successfully
- [ ] Documentation has been updated
- [ ] Release Notes have been updated

**Comment by garloff** [comment]
My DNS queries might reveal some details about my workload.
By using the default internal DNS from my operator (see #229), my queries are intermixed with queries from others; nevertheless the upstream DNS server could collect some data and patterns.
In order to judge my risk, I would like to have transparency on what upstream DNS servers are used.

**Comment by garloff** [comment]
This could be made transparent via self-descriptions.

**Comment by matfechner** [comment]
https://docs.openstack.org/neutron/latest/ovn/dhcp_opts.html

**Comment by anjastrunk** [comment]
> This could be made transparent via self-descriptions.

Good idea, I will add an issue in https://github.com/SovereignCloudStack/gx-credential-generator

**Comment by markus-hentsch** [comment]
> This could be made transparent via self-descriptions.

I share that sentiment. Based on my research in #229 I have not found a way to reliably retrieve this information via the API.

In OVS-based setups this is an internal setting (`dnsmasq_dns_servers`) that is not exposed. DHCP agents will use this setting to act a as an intermediate DNS resolver for tenant VMs.

For OVN-based setups it is a bit easier, since there is no agent between the tenant network and an upstream DNS. The information eventually ends up in `/etc/resolv.conf` entries of tenant VMs but can most likely not be retrieved earlier than that.
